services/management/jobs.go

```diff
@@ -0,0 +1,140 @@
+package management
+
+import (
+	"context"
+
+	"github.com/percona/pmm/api/managementpb"
+	"github.com/pkg/errors"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/services/agents"
+)
+
+// JobsAPIService provides methods for Jobs starting and management.
+type JobsAPIService struct {
+	db *reform.DB
+	r  *agents.Registry
+}
+
+// NewJobsAPIServer creates new jobs service.
+func NewJobsAPIServer(db *reform.DB, registry *agents.Registry) *JobsAPIService {
+	return &JobsAPIService{
+		db: db,
+		r:  registry,
+	}
+}
+
+// GetJob returns job result.
+func (s *JobsAPIService) GetJob(ctx context.Context, req *managementpb.GetJobRequest) (*managementpb.GetJobResponse, error) {
+	result, err := models.FindJobResultByID(s.db.Querier, req.JobId)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &managementpb.GetJobResponse{
+		JobId:      result.ID,
+		PmmAgentId: result.PMMAgentID,
+		Done:       result.Done,
+	}
+
+	if !result.Done {
+		return resp, nil
+	}
+
+	if result.Error != "" {
+		resp.Result = &managementpb.GetJobResponse_Error_{
+			Error: &managementpb.GetJobResponse_Error{
+				Message: result.Error,
+			},
+		}
+
+		return resp, nil
+	}
+
+	switch result.Type {
+	case models.Echo:
+		echoResult, err := result.GetEchoJobResult()
+		if err != nil {
+			return nil, err
+		}
+
+		resp.Result = &managementpb.GetJobResponse_Echo_{
+			Echo: &managementpb.GetJobResponse_Echo{
+				Message: echoResult.Message,
+			}}
+	default:
+		return nil, errors.Errorf("Unexpected job type: %s", result.Type)
+	}
+
+	return resp, nil
+}
+
+// StartEchoJob starts echo job. Its purpose is testing.
+func (s *JobsAPIService) StartEchoJob(ctx context.Context, req *managementpb.StartEchoJobRequest) (*managementpb.StartEchoJobResponse, error) {
```

Comment:
Probably we shouldn't have public API to start Jobs, jobs will be called from a different places. like in `/backup/PerformBackup` and e.t.c.,

Reply:
And most of the code from this struct should be moved to new JobsService or JobsRegistry.

Reply:
Done, Jobs API moved to separate package with `beta` postfix